core/unix/signal.c

```diff
@@ -151,8 +151,13 @@ sig_is_alarm_signal(int sig)
 #define APP_HAS_SIGSTACK(info) \
     ((info)->app_sigstack.ss_sp != NULL && (info)->app_sigstack.ss_flags != SS_DISABLE)
 
+// Under normal circumstances the app_sigaction is always fully initialized,
+// but during detach there are points where we are still sending/receiving
+// signals after app_sigaction has been set to zeros.
```

Comment:
app_sigaction[sig] is actually lazily allocated only when the app has a handler, but this query would only happen when delivering a signal to the app's handler, which normally would only happen when the app has a handler, except during detach.  (Though to be defensive it is good to have the check anyway in case there's some other query source.)

Reply:
Thanks for the more info, I've updated the comment. ptal